yarpcerrors/errors.go

```diff
@@ -26,6 +26,19 @@ import (
 	"fmt"
 )
 
+// Wrap returns a new Status that wraps the original error.
+//
+// The Code should never be CodeOK, if it is, this will return nil.
+func Wrap(code Code, err error) *Status {
+	if code == CodeOK {
+		return nil
+	}
+	return &Status{
```

Comment:
Did you on purpose not check if the given error was not a `Status`? For instance, before in the method `FromError` we have
```
	if st, ok := fromError(err); ok {
		return st
	}
```

Reply:
That is an interesting point. Do you think we should allow wrapping a status (e.g. not found) in another status (e.g. invalid argument)?

Reply:
If I use `fromError` in the `Wrap`:

- If the `Status` is not the top-most error, we will loose part of the error chain. This is also true for `FromError`.
- If there is a `Status` in the chain, the `error` returned by `Wrap` will have a different status from what the caller has specified, e.g. the following test would _fail_:
```
var statusErr error = Newf(CodeAborted, "some error")
err := FromErrorWithCode(statusErr, CodeInvalidArgument) // the inner CodeAborted overrides the consumer's error code, which is kind of a surprise. 
assert.True(t, IsInvalidArgument(err))
```